forte/processors/allennlp_processors.py

```diff
@@ -0,0 +1,145 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+from allennlp.predictors import Predictor
+from texar.torch import HParams
+
+from ft.onto.base_ontology import Token, Sentence, Dependency
+from forte.common.resources import Resources
+from forte.data import DataPack
+from forte.processors.base import PackProcessor
+from forte.common import ProcessorConfigError
+
+logger = logging.getLogger(__name__)
+
+__all__ = [
+    "AllenNLPProcessor",
+]
+
+# pylint: disable=line-too-long
+MODEL2URL = {
+    'stanford_dependencies': "https://allennlp.s3.amazonaws.com/models/biaffine-dependency-parser-ptb-2018.08.23.tar.gz",
+    'universal_dependencies': "https://allennlp.s3.amazonaws.com/models/biaffine-dependency-parser-ud-2018.08.23.tar.gz",
+}
+# pylint: enable=line-too-long
+
+
+class AllenNLPProcessor(PackProcessor):
+
+    # pylint: disable=attribute-defined-outside-init,unused-argument
+    def initialize(self, resource: Resources, configs: HParams):
+        self.processors = configs.processors
+        if self.processors is None or self.processors == "":
+            self.processors = self.default_configs()['processors']
+
+        if configs.output_format not in MODEL2URL:
+            raise ProcessorConfigError('Incorrect value for output_format')
+        model_url = MODEL2URL[configs.output_format]
+        self.predictor = Predictor.from_path(model_url)
+
+        self.overwrite_entries = configs.overwrite_entries
+        self.allow_parallel_entries = configs.allow_parallel_entries
+        if self.overwrite_entries:
+            logger.warning("`overwrite_entries` is set to True. Deleting an "
+                           "annotation doesn't guarantee deletion of all the "
+                           "associated links.")
+            if self.allow_parallel_entries:
+                logger.warning('Both `overwrite_entries` and '
+                               '`allow_parallel_entries` are True, all '
+                               'existing conflicting entries will be deleted.')
+        else:
+            if not self.allow_parallel_entries:
+                logger.warning('Both `overwrite_entries` and '
+                               '`allow_parallel_entries` are False, '
+                               'processor will only run if there are no '
+                               'existing conflicting entries.')
+
+    @staticmethod
+    def default_configs():
+        """
+        This defines a basic Hparams structure for AllenNLP.
+        :return:
+        """
+        return {
+            'processors': "tokenize,pos,depparse",
+            # format of the POS tags and dependency parsing
+            'output_format': "universal_dependencies",
```

Comment:
Can allennlp have other dependency type?

Reply:
There are two pre-trained models which I could see on their Github: stanford and universal